app/scripts/modules/core/src/managed/Environments.tsx

```diff
@@ -52,6 +56,38 @@ export function Environments({ app }: IEnvironmentsProps) {
     [selectedArtifact, artifacts],
   );
 
+  if (loadFailure) {
+    return (
+      <div style={{ width: '100%' }}>
+        <h4 className="text-center">There was an error loading environments.</h4>
```

Comment:
I instinctively used the application refresher and I think I would prefer to drive people to use that instead of having refreshes sprinkled throughout the UI?

Reply:
🤷‍♂ ok — i was pretty ambivalent about it.